doc/contributing/docs/terms.rst

```diff
@@ -0,0 +1,119 @@
+Using Terminology
+=================
+
+Concepts and Terms
+------------------
+
+To write well about some subject matter, one needs to know the objects of this subject matter
+and use the right, carefully selected words for them.
+Such objects are called concepts, and the words for them are called terms.
+
+..  glossary::
+
+    concept
+        Concept is the the idea of some object, attribute or action.
+        It is independent of languages, audience and products, it just exists.
+
+        For example, a large database can be partitioned into smaller instances,
+        which are easier to operate and together can exceed the throughput of
+        a single large database instance. Such instances can exchange data
+        to keep it consistent between them.
+
+    term
+        Term is a word, which authors or a particular book, article or documentation set
+        have explicitly selected to denote a :term:`concept` in a particular language
+        and for a particular audience.
+
+        For example, in Tarantool we use the term "[database] sharding" to denote the type of database
+        partitioning mentioned earlier.
+
+Use Preferred Terms
+-------------------
+
+The purpose of using terms is to write concisely and unambiguously,
+which is good for readers.
+But selecting terms is hard.
+Often there are two or more terms for one concept, popular with the community,
+so there's no obvious choice.
+Indeed, selecting and consistently using any of them is much better,
+than not selecting and just using random term each time.
+
+..  glossary::
+
+    restricted term
+        A restricted term is a word which authors have explicitly forbidden to use for denoting a :term:`concept`.
+        Such word is sometimes used as a :term:`term` for the same :term:`concept` elsewhere:
+        in the community, in other books, or in other products' documentation.
+        Sometimes, this word is used to denote a similar, but different concept.
+        In this case the right choice of terms helps us differentiate between concepts.
+
+        For example, in Tarantool we don't use the term "[database] segmentation"
+        to denote what we call "database sharding".
+        Although, other authors could do so.
+        Also, there is a term "[database] partitioning" which we use to denote
+        a wider concept, which is different from sharding.
+
+Define Terms
+------------
+
+Define the term by explaining the concept.
+For key concepts and the ones unique to Tarantool,
+the definition should always be in our documentation.
+
+    Database sharding is a type of horizontal partitioning.
+
+To define a term, use the ``glosary`` directive in the following way:
```

Comment:
glossary

Reply:
Maybe it's a good place to explain if it should be a separate page in a documentation project where we collect all of the terms, or it can be used in some other way.